src/data/sym/basic.lean

```diff
@@ -324,36 +326,45 @@ protected def cast {n m : ℕ} (h : n = m) : sym α n ≃ sym α m :=
 
 @[simp] lemma cast_rfl : sym.cast rfl s = s := subtype.ext rfl
 
-@[simp] lemma cast_cast {n n' n'' : ℕ} (s : sym α n) (h : n = n') (h' : n' = n'') :
+@[simp] lemma cast_cast {n'' : ℕ} (h : n = n') (h' : n' = n'') :
   sym.cast h' (sym.cast h s) = sym.cast (h.trans h') s := rfl
 
-@[simp] lemma coe_cast {n m : ℕ} (h : n = m) (s : sym α n) :
-  (sym.cast h s : multiset α) = s := rfl
+@[simp] lemma coe_cast  (h : n = m) : (sym.cast h s : multiset α) = s := rfl
+
+@[simp] lemma mem_cast (h : n = m) : a ∈ sym.cast h s ↔ a ∈ s := iff.rfl
 
 /-- Append a pair of `sym` terms. -/
-def append {n n' : ℕ} (s : sym α n) (s' : sym α n') : sym α (n + n') :=
+def append (s : sym α n) (s' : sym α n') : sym α (n + n') :=
 ⟨s.1 + s'.1, by simp_rw [← s.2, ← s'.2, map_add]⟩
 
-@[simp] lemma append_inj_right {n n' : ℕ} (s : sym α n) (t t' : sym α n') :
+@[simp] lemma append_inj_right (t t' : sym α n') :
   s.append t = s.append t' ↔ t = t' :=
 subtype.ext_iff.trans $ (add_right_inj _).trans subtype.ext_iff.symm
 
-@[simp] lemma append_inj_left {n n' : ℕ} (s s' : sym α n) (t : sym α n') :
+@[simp] lemma append_inj_left (s s' : sym α n) (t : sym α n') :
   s.append t = s'.append t ↔ s = s' :=
 subtype.ext_iff.trans $ (add_left_inj _).trans subtype.ext_iff.symm
 
-lemma append_comm {n n' : ℕ} (s : sym α n) (s' : sym α n') :
+lemma append_comm (s' : sym α n') :
   s.append s' = sym.cast (add_comm _ _) (s'.append s) :=
 by { ext, simp [append, add_comm], }
 
+@[simp] lemma coe_append (s' : sym α n') : (s.append s' : multiset α) = s + s' := rfl
+
+lemma append_mem (s' : sym α m) : a ∈ s.append s' ↔ a ∈ s ∨ a ∈ s' := multiset.mem_add
```

Comment:
```suggestion
lemma mem_append_iff (s' : sym α m) : a ∈ s.append s' ↔ a ∈ s ∨ a ∈ s' := multiset.mem_add
```

Reply:
The multiset and list versions don't have the "iff" though. Should it just be `mem_append`?

Reply:
@alreadydone I think the reason for `list.mem_append` is that the lemmas with `a ∈ s.append s'` as the conclusion are have a couple directional variants: `list.mem_append_left` and `list.mem_append_right`. That's to say, `list.mem_append` is freed up and it's safe to drop the `iff`. If you want, you could ask Zulip for whether `sym.mem_append_iff` or `sym.mem_append` is preferred. I personally think having the `iff` is more informative at little cost, but both should be fine.